include/seqan3/search/algorithm/search.hpp

```diff
@@ -163,7 +200,7 @@ inline auto search(std::initializer_list<char const * const> const & queries,
     std::vector<std::string_view> query;
     query.reserve(std::ranges::size(queries));
     std::ranges::for_each(queries, [&query] (char const * const q) { query.push_back(std::string_view{q}); });
-    return search(query, index, cfg);
+    return search(std::move(query) | views::persist, index, cfg);
```

Comment:
In general I am in favor of removing this interface altogether. For now, this is a quick fix to propagate the temporary as a view to the result range.

Reply:
the result range that is returned holds the queries. It requires the queries to be a view. Hence the initialiser list which is converted into a std::vector must propagate the queries on the heap to the result range. Just using view::all would cause a dangling reference.